python/core/additions/qgsfunction.py

```diff
@@ -99,78 +111,79 @@ def referencedColumns(self, node):
         def handlesNull(self):
             return self.handles_null
 
-    helptemplate = string.Template("""<h3>$name function</h3><br>$doc""")
-    name = kwargs.get('name', function.__name__)
-    helptext = kwargs.get('helpText') or function.__doc__ or ''
+    helptemplate = string.Template("<h3>$name function</h3><br>$doc")
+    name = kwargs.get("name", function.__name__)
+    helptext = kwargs.get("helpText") or function.__doc__ or ""
     helptext = helptext.strip()
-    expandargs = False
-
-    if arg_count == "auto":
-        # Work out the number of args we need.
-        # Number of function args - 2.  The last two args are always feature, parent.
-        args = inspect.getfullargspec(function).args
-        number = len(args)
-        arg_count = number - 2
-        if args[-1] == 'context':
-            arg_count -= 1
-        expandargs = True
-
-    register = kwargs.get('register', True)
+
+    register = kwargs.get("register", True)
     if register and QgsExpression.isFunctionName(name):
         if not QgsExpression.unregisterFunction(name):
             msgtitle = QCoreApplication.translate("UserExpressions", "User expressions")
-            msg = QCoreApplication.translate("UserExpressions",
-                                             "The user expression {0} already exists and could not be unregistered.").format(
-                name)
+            msg = QCoreApplication.translate(
+                "UserExpressions", "The user expression {0} already exists and could not be unregistered."
+            ).format(name)
             QgsMessageLog.logMessage(msg + "\n", msgtitle, Qgis.Warning)
             return None
 
     function.__name__ = name
     helptext = helptemplate.safe_substitute(name=name, doc=helptext)
-    f = QgsPyExpressionFunction(function, name, arg_count, group, helptext, usesgeometry, referenced_columns,
-                                expandargs, handlesnull)
 
-    # This doesn't really make any sense here but does when used from a decorator context
-    # so it can stay.
+    # Legacy: if args was not 'auto', parameters were passed as a list
+    params_as_list = kwargs.get("params_as_list", kwargs.get("args", "auto") != "auto")
+    f = QgsPyExpressionFunction(
+        function, name, group, helptext, usesgeometry, referenced_columns, handlesnull, params_as_list
+    )
+
     if register:
         QgsExpression.registerFunction(f)
     return f
 
 
-def qgsfunction(args='auto', group='custom', **kwargs):
+def qgsfunction(args="auto", group="custom", **kwargs):
     r"""
     Decorator function used to define a user expression function.
-
-    :param args: Number of parameters, set to 'auto' to accept a variable length of parameters.
+    :param args: DEPRECATED since QGIS 3.32. Use the "params_as_list" keyword argument instead if you want to pass parameters as a list.
```

Comment:
I'd remove the "instead". People might have set an `args` value for other reasons too (if only because they had no idea what it did :D ).

Reply:
Ok